### ISSUE TITLE
unable to add firebase package to project in xcode: authentication failed because no credentials were provided

### ISSUE THREAD

**Issue by mkandan** [firebase/firebase-ios-sdk#11304]
### Description

xcode: 14.3 (i even tried 14.2 and 14.1, but got the same error) macOS: 13.3.1 (a) (22E772610a)

tldr/video: https://www.loom.com/share/3ddee7d80007420592297c9421737ea5
similar to this issue https://github.com/firebase/firebase-ios-sdk/issues/7777

getting hit with an "authentication failed because no credentials were provided" while trying to add the firebase ios package to my project. driving me nuts!

have tried all combinations of user settings (clone using HTTPS and SSH) and SSH Key settings (id_ed25519 and id_rsa) -- this was recommended by various stack overflow threads

the token lets me commit/push/etc fine to my repo, but unable to add SPM packages

have rebooted mac, restarted XCode, Uninstalled and reinstalled XCode 14.3, then 14.2 and then tried 14.1 but it persists

### Reproducing the issue

1. make sure you have an Apple ID and GitHub accounts added to XCode
2. create new xcode project
3. follow along setup instructions on firebase.com
4. download and add GoogleService-Info.plist to XCode project folder
5. File - Add Packages
6. Search for `git@github.com:firebase/firebase-ios-sdk.git`, select exact version 10.9.0
7. Add Package
8. I tried various combinations of settings for my Github account in Xcode (Clone using HTTPS or SSH, and SSH Key - none, id_ed25519, id_rsa) but it would not work!

This should cause the errors described above (see above for Loom, below for Package.resolved error)

### Firebase SDK Version

10.9

### Xcode Version

14.3

### Installation Method

Swift Package Manager

### Firebase Product(s)

All

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Showing All Messages
PackageIndex.findPackages failed: featureDisabled


firebase-ios-sdk

Updating from git@github.com:firebase/firebase-ios-sdk.git
Updating from git@github.com:firebase/firebase-ios-sdk.git
git@github.com:firebase/firebase-ios-sdk.git: The source control operation failed because the revision “e88a0c9c82ed798e71ecb3e43e555f754b1c654c” could not be found. Make sure a valid revision exists in the repository and try again.



Resolving Package Graph Failed    5/17/23, 12:39 PM    5.6 seconds


```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by rizafran** [comment]
Thanks for reaching out, @mkandan. I was able to get the same error "_authentication failed because no credentials were provided_" when I followed your steps to repro. But when I tried to generate and add an SSH key, I was able to successfully add the SDK. With this, could you confirm if you have [generated](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) and [added the SSH key in your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)?

**Comment by mkandan** [comment]
Hey @rizafran!
> With this, could you confirm if you have [generated](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) and [added the SSH key in your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)?

Starting with generating an SSH, I followed the steps in the first link.

- Used the email address associated with my github account (the primary email listed in settings). 
- I saved this to the default location and overwrote the existing `.ssh/id_ed25519`
- The file saved, a public key was also saved, a SHA256 key fingerprint was outputted, and the key's randoart.

(The next step in the first article was ["Adding your SSH key to the ssh-agent"](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent). So I did that

My `.ssh/config` file's contents are
```

**Comment by mkandan** [comment]
Hey @rizafran (accidentally closed the issue, sorry for spam)!
> With this, could you confirm if you have [generated](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) and [added the SSH key in your GitHub account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)?

Starting with generating an SSH, I followed the steps in the first link.

- Used the email address associated with my github account (the primary email listed in settings). 
- I saved this to the default location and overwrote the existing `.ssh/id_ed25519`
- The file saved, a public key was also saved, a SHA256 key fingerprint was outputted, and the key's randoart.

(The next step in the first article was ["Adding your SSH key to the ssh-agent"](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent). So I did that

My `.ssh/config` file's contents are
```
Host *
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_ed25519

# Fig ssh integration. Keep at the bottom of this file.
Match all
  Include ~/.fig/ssh
```
Then added SSH private key to the ssh-agent and store your passphrase in the keychain, terminal responded with `Identity added: /Users/mani/.ssh/id_ed25519 (mani@**********.com)`

**And then I moved onto the second article.** to add the SSH key to my Github account via web browser.
Skipped down to the [Adding a new SSH key to your account](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account#adding-a-new-ssh-key-to-your-account) section.

I deleted all existing SSH keys (1 for my laptop (the one that existed before I overwrote it in the steps from the first article), 2 for XCode that were made while I was trying anything & everything to solve this issue). Then added in the copied key as a Authentication Key.
<img width="1460" alt="image" src="https://github.com/firebase/firebase-ios-sdk/assets/40218068/ae3f1620-4940-4aa2-9678-36acd575684b">


### Finally
- I quit XCode
- Relaunch XCode, and I remove my previously signed in Github Account from Settings - Account for good measure
- Delete existing Github Personal Access Tokens via web browser at this [page](https://github.com/settings/tokens)
- Retry adding Github Account, it requests a personal access token so I make a new one. It is a "Personal access tokens (classic)" type with (just for testing purposes) all permissions and no expiration.
- I copy the generated token, it starts with `ghp_...`. back to xcode now
- Create new project - multiplatform - App
Here is a video of the following process 
https://www.loom.com/share/a965896acf324c15ac9293c79ed73351
- Add my Github Account - username (mkandan) and the personal access token I just generated
- File - add packages - firebase
- errors!!!

### What if I search git@github.com:firebase/firebase-ios-sdk.git instead
https://www.loom.com/share/ca2c6c472cf84aea88c04e2979d409f7
- Then it lets me specify "Exact version" of 10.9.0
- I press add package
- It loads for a while

So here is the log file for that
[Add Package_2023-05-19T12-46-20.txt](https://github.com/firebase/firebase-ios-sdk/files/11520024/Add.Package_2023-05-19T12-46-20.txt)

---
So yea, I don't know what else to do. Any ideas?

**Comment by rizafran** [comment]
Thanks for sharing your steps to repro, @mkandan. I tried to reproduce it again on Xcode 14.3, but I'm still able to successfully install the SDK. Does the issue also appear in other machines or when you re-install the Xcode?

**Comment by rizafran** [comment]
To figure out if Firebase causes the issue, have you tried installing other SDKs? If so, does the issue persist?

**Comment by mkandan** [comment]
@rizafran Yes, the same issue was happening.

Update: I didn't want to spend time debugging through macOS and XCode, so I just wiped my laptop (was long overdue anyways) and reinstalled everything including XCode. Works fine now. I think it's safe to assume this was an issue on my end. Closing for now.
